http/basic/src/main/java/io/cloudevents/http/HttpMessageReader.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cloudevents.http;
+
+import io.cloudevents.SpecVersion;
+import io.cloudevents.core.message.impl.BaseGenericBinaryMessageReaderImpl;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import static io.cloudevents.http.CloudEventsHeaders.CE_PREFIX;
+
+public class HttpMessageReader extends BaseGenericBinaryMessageReaderImpl<String, String> {
+
+    private final Iterable<Map.Entry<String,String>> headers;
+
+    protected HttpMessageReader(SpecVersion version, Iterable<Map.Entry<String,String>> headers, byte[] body) {
```

Comment:
What about just `Map<String, String>` for the `headers` parameter? I think that would be friendlier for users. Obviously you can make the type here out of a `Map<String, String>` by calling `map.entrySet()` but that feels a bit more complicated than it needs to be.